### PR TITLE
Add 'phoenix-core` to a workspace called phoenix

### LINKED ISSUE #171: turn 'phoenix-core' into a 'phoenix' workspace
#### Summary
With the decision to move the transfer circuits out of 'rusk' and into the phoenix implementation library #170, we have decided to create a new workspace called 'phoenix' with the following structure:
```
phoenix
  |- core
  |- circuits
  |- docs
```

The first step and scope of this issue is to add the workspace and its members. This will also include adding the existing elgamal encryption into the 'circuits' member.

The next step is adding the actual transfer circuits and gadgets as part of #170

### PR DESCRIPTION
**Pull request by moCello**
Resolves #171

### CODE REVIEW

**Review by xevisalle — APPROVED** [review]
LPMGTM

File: circuits/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
+
+- Add `phoenix-circuits` as a workspace member of `pheonix` [#171]
```
Comment (HDauven) [inline circuits/CHANGELOG.md]: pheonix -> phoenix

File: circuits/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "phoenix-circuits"
+version = "0.1.0"
+edition = "2021"
+repository = "https://github.com/dusk-network/phoenix/circuits"
+description = "Circuit definitions for Phoenix, an anonymity-preserving ZKP-based transaction model"
```
Comment (HDauven) [inline circuits/Cargo.toml]: anonymity-preserving -> privacy-preserving

File: core/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+[package]
+name = "phoenix-core"
+version = "0.27.0"
+edition = "2021"
+repository = "https://github.com/dusk-network/phoenix/core"
+description = "Core types and functionalities for Phoenix, an anonymity-preserving ZKP-based transaction model"
```
Comment (HDauven) [inline core/Cargo.toml]: Anonymity-preserving -> privacy-preserving

File: core/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Phoenix Core
+
+Phoenix is the transaction model used by Dusk, an open-source public blockchain with a UTXO-based architecture that allows the execution of obfuscated transactions and confidential smart contracts.
```
Comment (HDauven) [inline core/README.md]: ...allows for the...

**Review by HDauven — APPROVED** [review]
LGTM, couple of small recommendations

**Pull request merged**